src/main/java/de/hysky/skyblocker/skyblock/dwarven/PowderMiningTracker.java

```diff
@@ -0,0 +1,339 @@
+
+		HudRenderEvents.AFTER_MAIN_HUD.register((context, tickCounter) -> {
+			if (Utils.getLocation() != Location.CRYSTAL_HOLLOWS || !isEnabled()) return;
+			int y = MinecraftClient.getInstance().getWindow().getScaledHeight() / 2 - 100;
+			var set = SHOWN_REWARDS.object2IntEntrySet();
+			for (Object2IntMap.Entry<Text> entry : set) {
+				context.drawTextWithShadow(MinecraftClient.getInstance().textRenderer, entry.getKey(), 5, y, 0xFFFFFF);
+				context.drawTextWithShadow(MinecraftClient.getInstance().textRenderer, Text.of(String.valueOf(entry.getIntValue())), 10 + MinecraftClient.getInstance().textRenderer.getWidth(entry.getKey()), y, 0xFFFFFF);
+				y += 10;
+			}
+			context.drawTextWithShadow(MinecraftClient.getInstance().textRenderer, Text.literal("Gain: " + NumberFormat.getInstance().format(profit) + " coins").formatted(Formatting.GOLD), 5, y + 10, 0xFFFFFF);
+		});
+
+		ItemPrice.ON_PRICE_UPDATE.register(() -> {
+			if (isEnabled()) recalculatePrices();
+		});
+
+		ClientLifecycleEvents.CLIENT_STARTED.register(PowderMiningTracker::loadRewards);
+		ClientLifecycleEvents.CLIENT_STOPPING.register(PowderMiningTracker::saveRewards);
+
+		//TODO: Sort out proper commands for this
+		ClientCommandRegistrationCallback.EVENT.register((dispatcher, registryAccess) -> dispatcher.register(
+				literal(SkyblockerMod.NAMESPACE)
+						.then(
+								literal("clearrewards")
+										.executes(context -> {
+											SHOWN_REWARDS.clear();
+											return 1;
+										})
+						)
+						.then(
+								literal("listrewards")
+										.executes(context -> {
+											var set = SHOWN_REWARDS.object2IntEntrySet();
+											for (Object2IntMap.Entry<Text> entry : set) {
+												MinecraftClient.getInstance().inGameHud.getChatHud().addMessage(entry.getKey().copy().append(" ").append(Text.of(String.valueOf(entry.getIntValue()))));
+											}
+											return 1;
+										})
+						)
+		));
+	}
+
+	private static void incrementReward(String itemName, String itemId, int amount) {
+		ALL_REWARDS.mergeInt(itemId, amount, Integer::sum);
+		if (!SkyblockerConfigManager.get().mining.crystalHollows.powderTrackerFilter.contains(itemName)) {
+			if (itemId.equals("GEMSTONE_POWDER")) {
+				SHOWN_REWARDS.merge(Text.literal("Gemstone Powder").formatted(Formatting.LIGHT_PURPLE), amount, Integer::sum);
+			} else {
+				ItemStack stack = ItemRepository.getItemStack(itemId);
+				if (stack == null) {
+					LOGGER.warn("Item stack for id `{}` is null! This might be caused by failed item repository downloads.", itemId);
+					return;
+				}
+				SHOWN_REWARDS.merge(stack.getName(), amount, Integer::sum);
+			}
+		}
+	}
+
+	private static int comparePriority(String s) {
+		s = GEMSTONE_SYMBOLS.matcher(s).replaceAll(""); // Removes the gemstone symbol from the string to make it easier to compare
+		// Puts gemstone powder at the top of the list, then gold and diamond essence, then gemstones by ascending rarity and then whatever else.
+		switch (s) {
```

Comment:
You can use pattern matching instead:
```java
return switch (s) {
	case "Gemstone Powder" -> 1;
	case "Gold Essence" -> 2;
	case "Diamond Essence" -> 3;
	case String s when s.startsWith("Rough") -> 4;
	case String s when s.startsWith("Flawed") -> 5;
	case String s when s.startsWith("Fine") -> 6;
	case String s when s.startsWith("Flawless") -> 7;
	default -> 8;
};
```

Reply:
Done in [`f207900`](https://github.com/SkyblockerMod/Skyblocker/pull/1065/commits/f207900045d7ef6a7f45004bef8b77c6c23a4465).